### ISSUE TITLE
Restore the Godot 3 "FileDialog.popup_hide" signal

### ISSUE THREAD

**Issue by Poobslag** [godotengine/godot-proposals#7084]
### Describe the project you are working on

A 2D puzzle game, being converted from Godot 3 to Godot 4

### Describe the problem or limitation you are having in your project

Godot 3 provided a FileDialog.popup_hide signal, Godot 4 provides no alternative for this. I have explored about 20-30 options but I could not find any combination of them which are sufficient -- the signals are emitted too frequently, not frequently enough, or prematurely. No combination of signals captures the case where the popup is hidden.

`FileDialog.dir_selected`, `FileDialog.file_selected` and `FileDialog.files_selected` are not emitted when the dialog is cancelled.

`AcceptDialog.canceled` is not emitted when the dialog is confirmed.

`AcceptDialog.confirmed` is not emitted when the dialog is cancelled. Additionally, it is emitted prematurely even if confirming does not result in the dialog being closed (for example if the user is prompted to overwrite a file)

`Window.close_requested` is not emitted when the dialog is dismissed with the cancel button.

`Window.visibility_changed` is emitted both when the dialog pops up and dismissed -- so the receiver of the signal must somehow get or maintain a reference to the dialog.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

A new `popup_hide` signal (or something similarly named) which works identically to its Godot 3 counterpart.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

`$FileDialog.popup_hide.connect(_on_FileDialog_popup_hide)`

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Yes, I have written a script which can hang off of any node to provide a `popup_hide` signal. It does this by listening for `visibility_changed` events, performing some boolean logic and then emitting a signal.

```
extends Node

signal popup_hide

# Called when the node enters the scene tree for the first time.
func _ready() -> void:
	get_parent().visibility_changed.connect(_on_Dialog_visibility_changed)


func _on_Dialog_visibility_changed() -> void:
	if not get_parent().visible:
		emit_signal("popup_hide")
```

### Is there a reason why this should be core and not an add-on in the asset library?

`popup_hide` is an obvious counterpart to `about_to_popup`. Either both signals should exist, or both signals should be removed -- it is unusual to be able to add a signal for one, but to require a custom 10-20 line script to implement the other.

**Comment by AThousandShips** [comment]
The signal is in 4.x though?

**Comment by AThousandShips** [comment]
Would involve either moving it up from `Popup` to `Window` or adding it specifically for the dialog class

**Comment by KoBeWi** [comment]
> popup_hide is an obvious counterpart to about_to_popup

That's not true. `about_to_popup` is called _before_ the popup appears, which allows to customize the popup's contents while it's still invisible.
